app.py

```diff
@@ -99,35 +114,41 @@ def stream_with_data(body, headers, endpoint):
             "messages": []
         }]
     }
-    with s.post(endpoint, json=body, headers=headers, stream=True) as r:
-        for line in r.iter_lines(chunk_size=10):
-            if line:
-                lineJson = json.loads(line.lstrip(b'data:').decode('utf-8'))
-                response["id"] = lineJson["id"]
-                response["model"] = lineJson["model"]
-                response["created"] = lineJson["created"]
-                response["object"] = lineJson["object"]
-
-                role = lineJson["choices"][0]["messages"][0]["delta"].get("role")
-                if role == "tool":
-                    response["choices"][0]["messages"].append(lineJson["choices"][0]["messages"][0]["delta"])
-                elif role == "assistant": 
-                    response["choices"][0]["messages"].append({
-                        "role": "assistant",
-                        "content": ""
-                    })
-                else:
-                    deltaText = lineJson["choices"][0]["messages"][0]["delta"]["content"]
-                    if deltaText != "[DONE]":
-                        response["choices"][0]["messages"][1]["content"] += deltaText                
-
-                yield json.dumps(response)
+    try:
+        with s.post(endpoint, json=body, headers=headers, stream=True) as r:
+            for line in r.iter_lines(chunk_size=10):
+                if line:
+                    lineJson = json.loads(line.lstrip(b'data:').decode('utf-8'))
+                    if 'error' in lineJson:
+                        yield json.dumps(lineJson) + "<newline>"
+                    response["id"] = lineJson["id"]
+                    response["model"] = lineJson["model"]
+                    response["created"] = lineJson["created"]
+                    response["object"] = lineJson["object"]
+
+                    role = lineJson["choices"][0]["messages"][0]["delta"].get("role")
+                    if role == "tool":
+                        response["choices"][0]["messages"].append(lineJson["choices"][0]["messages"][0]["delta"])
+                    elif role == "assistant": 
+                        response["choices"][0]["messages"].append({
+                            "role": "assistant",
+                            "content": ""
+                        })
+                    else:
+                        deltaText = lineJson["choices"][0]["messages"][0]["delta"]["content"]
+                        if deltaText != "[DONE]":
+                            response["choices"][0]["messages"][1]["content"] += deltaText                
+
+                    yield json.dumps(response) + "<newline>"
```

Comment:
Would it work to do `json.dumps(response).replace("\n", "\\n") + "\n"` .... I'm concerned if the response happens to contain "<newline>" somewhere in it.  I'd think all actual newlines returned by json.dumps must be in strings, so replacing it with "\n" should keep it functionally identical.